### ISSUE TITLE
Using python from conda installation

### ISSUE THREAD

**Issue by CarlAndersson** [sublimelsp/LSP-pylsp#93]
Is there any way to use this with conda on windows 10? I cannot figure out how to properly configure the python path so that the server can start.
Currently I get an error message:

> Failed to start LSP-pylsp - disabling for this window for the duration of
the current session.
Re-enable by running "LSP: Enable Language Sever In Project" from the command palette.
>
> --- Error: ---
'python' is not recognized as an internal or external command,
operable program of batch file.

**Comment by rchl** [comment]
Does this help maybe: https://github.com/sublimelsp/LSP-pylsp#virtual-environments ?

**Comment by CarlAndersson** [comment]
It changes the error message to 

> Failed to start LSP-pylsp - disabling for this window for the duration of
> the current session.
> Re-enable by running "LSP: Enable Language Sever In Project" from the command palette.
>
> --- Error ---
> Python was not found; run without arguments to install from the
> Microsoft Store, or disable this shortcut from Settings > Manage App
> Execution Aliases.

I've also tried to set the path in `LSP-pylsp.sublime-settings` with various options that seemed relevant.
I tried lots of combinations of these settings, either pointing to an executable or just the environment directory. 
```
{
	"command": ["C:\\Users\\username\\Miniconda3\\envs\\default\\Scripts\\pylsp.exe"],
	"env": {
        "PYTHONPATH": "C:\\Users\\username\\Miniconda3\\envs\\default\\python.exe",
        "MYPYPATH": "C:\\Users\\username\\Miniconda3\\envs\\default\\python.exe",
	},
	"settings": {
		"pylsp.plugins.jedi.environment": "C:\\Users\\username\\Miniconda3\\envs\\default"
	}
}
```

**Comment by rwols** [comment]
You also need “python.exe” in %PATH%

**Comment by CarlAndersson** [comment]
Adding python to my global user path seems to help that part, Now the language server starts!
However, I don't seem to get e.g. any hover info for functions, not even functions from the standard library.
Looking at the LSP log panel it seems like some tool (perhaps jedi) is looking for a python.exe at `path\to\env\Scripts\Python.exe`. Conda seems to put the executable at `path\to\env\Python.exe`. 
Is there some nice setting to fix this?

It's somewhat inconvenient for all parties that conda and venv isn't putting the python executable in the same place, but that's not really related to this tool.

**Comment by CarlAndersson** [comment]
I did some more digging and found a solution. At some point the value set in `pylsp.plugins.jedi.enrvironment` ends up in [`create_environment`](https://github.com/davidhalter/jedi/blob/66c52b4bc7cbe6db7e7f308ab73bd50b3de1073b/jedi/api/environment.py#L364)  which supports using a path to an executable instead of a "normal" environment. So I used the path to the `python.exe` file in my sublime settings instead. That seems to work good, at least at brief inspection.

I'm closing this since it's working for me now, and hopefully helps someone else using conda in the future.
Thanks for all the help and the very quick responses!
